src/objectivec/include/ort_genai_objc.h

```diff
@@ -0,0 +1,406 @@
+
+/**
+ * Encode text to sequences
+ *
+ * @param str The text to be encoded.
+ * @param error Optional error information set if an error occurs.
+ * @return The encoding result, or nil if an error occurs.
+ */
+- (nullable OGASequences*)encode:(NSString*)str
+                           error:(NSError**)error;
+
+/**
+ * Decode sequences to text
+ *
+ * @param data The sequences data to be encoded.
+ * @param error Optional error information set if an error occurs.
+ * @return The decoding result, or nil if an error occurs.
+ */
+- (nullable NSString*)decode:(OGAInt32Span*)data
+                       error:(NSError**)error;
+
+@end
+
+/**
+ * A tokenizer stream is to decoded token strings incrementally, one token at a time.
+ */
+@interface OGATokenizerStream : NSObject
+- (instancetype)init NS_UNAVAILABLE;
+
+/**
+ * Creates a tokenizer stream with an underlying tokenizer.
+ *
+ * @param tokenizer The underlying tokenizer to use.
+ * @param error Optional error information set if an error occurs.
+ * @return The instance, or nil if an error occurs.
+ */
+- (nullable instancetype)initWithTokenizer:(OGATokenizer*)tokenizer
+                                     error:(NSError**)error NS_DESIGNATED_INITIALIZER;
+
+/**
+ * Creates a tokenizer stream with a multi modal processor
+ *
+ * @param processor The underlying processor to use.
+ * @param error Optional error information set if an error occurs.
+ * @return The instance, or nil if an error occurs.
+ */
+- (nullable instancetype)initWithMultiModalProcessor:(OGAMultiModalProcessor*)processor
+                                               error:(NSError**)error NS_DESIGNATED_INITIALIZER;
+/**
+ * Decode one token.
+ *
+ * @param token The token to be decoded.
+ * @param error Optional error information set if an error occurs.
+ * @return The decoding result, or nil if an error occurs.
+ */
+- (nullable NSString*)decode:(int32_t)token
+                       error:(NSError**)error;
+@end
+
+/**
+ * Wraps a raw int32_t pointer and its size. Represents a data sequence.
+ */
+@interface OGAInt32Span : NSObject
+
+- (instancetype)init NS_UNAVAILABLE;
+/**
+ * Creates a span with underlying data.
+ *
+ * @param pointer The underlying data pointer to use.
+ * @param size The underlying data size.
+ * @return The instance, or nil if an error occurs.
+ */
+- (nullable instancetype)initWithDataPointer:(const int32_t*)pointer size:(size_t)size;
+
+/**
+ * The underlying data pointer
+ */
+- (const int32_t*)pointer;
+/**
+ * The underlying data size
+ */
+- (size_t)size;
+/**
+ * The last element in this data sequence
+ * @param error Optional error information set if an error occurs.
+ * @return The last element, or nil if an error occurs.
+ */
+- (int32_t)lastElementWithError:(NSError**)error NS_SWIFT_NAME(lastElement());
+
+@end
+
+/**
+ * Wraps a raw int64_t pointer and its size. Represents a data sequence.
+ */
+@interface OGAInt64Span : NSObject
```

Comment:
do we still need `OGAInt64Span`?

Reply:
yeah we don't. will move it into `oge_internal.h` and out of public interfaces.